common/static/common/js/discussion/discussion_router.js

```diff
@@ -89,16 +100,44 @@
                 if (this.thread) {
                     return this.nav.setActiveThread(this.thread.get("id"));
                 } else {
-                    return this.nav.goHome;
+                    return this.nav.goHome();
```

Comment:
@bjacobel Yes and no. Yes - the intent here was to show discussion main page

![image](https://cloud.githubusercontent.com/assets/3330048/16683001/2c6d44b2-4506-11e6-89e9-1b0eb11147b1.png)

However, goHome is a method, and setActiveThread is called from Backbone event handler that  does not expect to get anything returned, so return value is never used.

On the other hand, the problem was masked by some code that is executed later - it triggered `allThreads` method above, which executes `goHome`, effectively masking the problem.

One more thing - looks like this else clause is a "almost" dead code anyway, as it is only executed during initialization (`this.thread` is never unset, even after thread is deleted). and it basically did nothing for years and no one noticed. I'll just remove from here.